hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/table/HoodieTableSource.java

```diff
@@ -334,6 +334,9 @@ private List<MergeOnReadInputSplit> buildInputSplits() {
     HoodieTableFileSystemView fsView = new HoodieTableFileSystemView(metaClient,
         // file-slice after pending compaction-requested instant-time is also considered valid
         metaClient.getCommitsAndCompactionTimeline().filterCompletedAndCompactionInstants(), fileStatuses);
+    if (!fsView.getLastInstant().isPresent()) {
+      throw new HoodieException("No completed instant found for table : " + conf.getString(FlinkOptions.TABLE_NAME));
+    }
```

Comment:
Instead of throwing exception, can we returns a empty list so the query result is just empty?